src/binary/raw_binary_reader.rs

```diff
@@ -228,6 +228,8 @@ where
  */
 #[derive(Clone, Debug)]
 pub struct CursorState {
+    // Indicates whether the reader is positioned over an IVM, a value, or nothing.
+    current_item: RawStreamItem,
```

Comment:
PR tour: the raw binary reader now tracks its current `RawStreamItem` (discovered during calls to `next()`) in case the user later requests it with `reader.current()`.

This is analogous to the way ion-java's user-level `IonReader` has a [`IonType next()`](https://www.javadoc.io/doc/com.amazon.ion/ion-java/latest/com/amazon/ion/IonReader.html#next--) and an [`IonType getType()`](https://www.javadoc.io/doc/com.amazon.ion/ion-java/latest/com/amazon/ion/IonReader.html#getType--), but allows for stream item types other than `Value`.